packages/fxa-settings/src/lib/integrations/integration-factory-flags.ts

```diff
@@ -2,9 +2,142 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { DefaultRelierFlags, RelierFlags } from '../reliers';
+import { Constants } from '../constants';
+import { ModelDataStore, UrlQueryData } from '../model-data';
+import { IntegrationFlags } from '../integrations/interfaces';
 
-// TODO: Using Relier flags is temporary, we will combine them in FXA-7308
-export class DefaultIntegrationFlags
-  extends DefaultRelierFlags
-  implements RelierFlags {}
+const DEVICE_PAIRING_SUPPLICANT_PATHNAME_REGEXP = /^\/pair\/supp/;
+
+/**
+ * Extrapolates flags from the state of the current data store. The collective state of these flags are used by
+ * the factory to determine what underlying type of relier to create.
+ *
+ * Note: this logic was ported from fxa-content-server app-start.js.
+ */
+export class DefaultIntegrationFlags implements IntegrationFlags {
+  protected get pathname() {
+    return this.urlQueryData.pathName;
+  }
+
+  constructor(
+    private urlQueryData: UrlQueryData,
+    private storageData: ModelDataStore
+  ) {}
+
+  isDevicePairingAsAuthority() {
+    return (
+      this.searchParam('redirect_uri') ===
```

Comment:
Just realized we could make models for accessing these things if we wanted to.

Reply:
I think I would prefer we use type guards and just check the `integration.type`. But yeah, if we need it in the future, 👍